src/cache/core/types/Cache.ts

```diff
@@ -104,12 +161,197 @@ export declare namespace Cache {
     ) => any;
   }
 
-  export import DiffResult = DataProxy.DiffResult;
-  export import ReadQueryOptions = DataProxy.ReadQueryOptions;
-  export import ReadFragmentOptions = DataProxy.ReadFragmentOptions;
-  export import WriteQueryOptions = DataProxy.WriteQueryOptions;
-  export import WriteFragmentOptions = DataProxy.WriteFragmentOptions;
-  export import UpdateQueryOptions = DataProxy.UpdateQueryOptions;
-  export import UpdateFragmentOptions = DataProxy.UpdateFragmentOptions;
-  export import Fragment = DataProxy.Fragment;
+  export interface ReadQueryOptions<
+    TData,
+    TVariables extends OperationVariables,
+  > {
+    /**
+     * The GraphQL query shape to be used constructed using the `gql` template
+     * string tag from `graphql-tag`. The query will be used to determine the
+     * shape of the data to be read.
+     */
+    query: DocumentNode | TypedDocumentNode<TData, TVariables>;
+
+    /**
+     * Any variables that the GraphQL query may depend on.
+     */
+    variables?: TVariables;
+
+    /**
+     * The root id to be used. Defaults to "ROOT_QUERY", which is the ID of the
+     * root query object. This property makes writeQuery capable of writing data
+     * to any object in the cache.
+     */
+    id?: string;
+    /**
+     * Whether to return incomplete data rather than null.
+     * Defaults to false.
+     */
+    returnPartialData?: boolean;
+    /**
+     * Whether to read from optimistic or non-optimistic cache data. If
+     * this named option is provided, the optimistic parameter of the
+     * readQuery method can be omitted. Defaults to false.
+     */
+    optimistic?: boolean;
+  }
+
+  export interface ReadFragmentOptions<
+    TData,
+    TVariables extends OperationVariables,
+  > {
+    /**
+     * The root id to be used. This id should take the same form as the
+     * value returned by your `dataIdFromObject` function. If a value with your
+     * id does not exist in the store, `null` will be returned.
+     */
+    id?: string;
+
+    /**
+     * A GraphQL document created using the `gql` template string tag from
+     * `graphql-tag` with one or more fragments which will be used to determine
+     * the shape of data to read. If you provide more than one fragment in this
+     * document then you must also specify `fragmentName` to select a single.
+     */
```

Comment:
```suggestion
    /**
     * A GraphQL document created using the `gql` template string tag
     * with one or more fragments which will be used to determine
     * the shape of data to read. If you provide more than one fragment in this
     * document then you must also specify `fragmentName` to select a single.
     */
```

Reply:
```suggestion
    /**
     * A GraphQL document created using the `gql` template string tag from
     * `graphql-tag` with one or more fragments which will be used to determine
     * the shape of data to read. If you provide more than one fragment in this
     * document then you must also specify `fragmentName` to specify which 
     * fragment is the root fragment.
     */
```

The last part of the last sentence sounds incomplete so I'd suggest this change.